client/ayon_silhouette/api/pipeline.py

```diff
@@ -0,0 +1,374 @@
+import os
+import logging
+import contextlib
+from functools import partial
+
+import pyblish.api
+from qtpy import QtCore
+
+from ayon_core.host import HostBase, IWorkfileHost, ILoadHost, IPublishHost
+from ayon_core.tools.utils import host_tools
+from ayon_core.pipeline import (
+    register_loader_plugin_path,
+    register_creator_plugin_path,
+    # AYON_CONTAINER_ID,
+    AYON_INSTANCE_ID,
+    get_current_context,
+)
+from ayon_core.pipeline.load import any_outdated_containers
+from ayon_core.lib import emit_event, register_event_callback
+from ayon_core.pipeline.context_tools import (
+    version_up_current_workfile,
+    get_current_task_entity
+)
+from ayon_core.settings import get_current_project_settings
+from .workio import (
+    open_file,
+    save_file,
+    file_extensions,
+    has_unsaved_changes,
+    current_file
+)
+from . import lib
+
+import fx
+import hook
+
+log = logging.getLogger("ayon_silhouette")
+
+HOST_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+PLUGINS_DIR = os.path.join(HOST_DIR, "plugins")
+PUBLISH_PATH = os.path.join(PLUGINS_DIR, "publish")
+LOAD_PATH = os.path.join(PLUGINS_DIR, "load")
+CREATE_PATH = os.path.join(PLUGINS_DIR, "create")
+INVENTORY_PATH = os.path.join(PLUGINS_DIR, "inventory")
+
+AYON_CONTAINERS = lib.AYON_CONTAINERS
+AYON_CONTEXT_CREATOR_IDENTIFIER = "io.ayon.create.context"
+
+
+def defer(callable, timeout=0):
+    """Defer a callable to the next event loop."""
+    QtCore.QTimer.singleShot(timeout, callable)
+
+
+class SilhouetteHost(HostBase, IWorkfileHost, ILoadHost, IPublishHost):
+    name = "silhouette"
+
+    def __init__(self):
+        super(SilhouetteHost, self).__init__()
+
```

Comment:
```suggestion
```

Reply:
Yup - was just about to remove this :) Mostly came from copying from other hosts. 👍